e2etest/container/container.go

```diff
@@ -263,19 +262,20 @@ func randomAvailablePort(t *testing.T) int {
 	// Base port and spread range for port selection
 	const (
 		basePort  = 20000
-		portRange = 20000
+		portRange = 30000
 	)
 
-	// Seed the random number generator to ensure randomness
-	rand.Seed(time.Now().UnixNano())
-
 	// Try up to 10 times to find an available port
 	for i := 0; i < 10; i++ {
 		port := randPort(basePort, portRange)
 		listener, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", port))
 		if err != nil {
 			continue
 		}
+
+		// listen on a port for a bit before closing, this should be just enough time for when other tests start
+		// when they try to net.Listen they should get an err and try a different port
+		time.Sleep(200 * time.Millisecond)
```

Comment:
this seems hacky... 
I would feel better by setting a specific port range for each test, but also doesn't seem too good 
Any other idea to solve this? @babylonlabs-io/core-dev

Reply:
Right, but tests do not share state, so having a mutex or a map here doesn't help. One way would be to have a test which would be ran first (e.g called 01Setup and without parallel flag), which would create a file and a lockfile, then all other parallel tests startup and read/write port that they are acquiring to this shared file (locking and unlocking).
I feel like this is a bit much, but if we start having collisions I would implement it.
Thoughts @KonradStaniec @RafilxTenfen

Reply:
Also reason for needing to have a test called 01Setup (or something like that), is that because all tests start in parallel so they would also be having race condition for the initial file creation

Reply:
Not super hard to implement, I'll try it, see if it works and is worth the extra code. Thorugh this was clever and a nice hack :D

Reply:
> Right, but tests do not share state, so having a mutex or a map here doesn't help.

They do not share it now but we could introduce it. You could create file `port_map.go` in our `e2etest` package have there `var port_map = map[int]bool` and protect access to it. It seems simpler that having file with lock file.

Or do I miss something about how go shares state between tests ?